src/transformers/modeling_utils.py

```diff
@@ -1160,12 +1160,15 @@ def num_parameters(self, only_trainable: bool = False, exclude_embeddings: bool
                 # For 4bit models, we need to multiply the number of parameters by 2 as half of the parameters are
                 # used for the 4bit quantization (uint8 tensors are stored)
                 if is_loaded_in_4bit and isinstance(param, bnb.nn.Params4bit):
-                    quant_storage = self.hf_quantizer.quantization_config.bnb_4bit_quant_storage
-                    # For compatibility with older PT version - see: https://github.com/huggingface/peft/pull/1635
-                    nb_params = (
-                        quant_storage.itemsize if hasattr(quant_storage, "itemsize") else quant_storage.element_size()
+                    if hasattr(param, "element_size"):
```

Comment:
Why not simply use:

```py
quant_storage = self.hf_quantizer.quantization_config.bnb_4bit_quant_storage
num_bytes = quant_storage.element_size()
```

`element_size` seems present from 1.9.1: https://pytorch.org/docs/1.9.1/search.html?q=element_size&check_keywords=yes&area=default to latest: https://pytorch.org/docs/2.2/search.html?q=element_size&check_keywords=yes&area=default

Reply:
cc @winglian @BenjaminBossan @hiyouga

Reply:
`self.hf_quantizer.quantization_config.bnb_4bit_quant_storage` is a `torch.dtype` instance while only the `torch.tensor` has `element_size()`